flink-table-store-core/src/main/java/org/apache/flink/table/store/table/sink/TableCommit.java

```diff
@@ -18,121 +18,55 @@
 
 package org.apache.flink.table.store.table.sink;
 
-import org.apache.flink.table.store.file.manifest.ManifestCommittable;
-import org.apache.flink.table.store.file.operation.FileStoreCommit;
-import org.apache.flink.table.store.file.operation.FileStoreExpire;
+import org.apache.flink.table.store.annotation.Experimental;
+import org.apache.flink.table.store.file.catalog.CatalogLock;
 import org.apache.flink.table.store.file.operation.Lock;
-import org.apache.flink.table.store.file.operation.PartitionExpire;
+import org.apache.flink.table.store.table.Table;
 
-import javax.annotation.Nullable;
-
-import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
+import java.util.Set;
 
 /**
- * An abstraction layer above {@link FileStoreCommit} and {@link FileStoreExpire} to provide
- * snapshot commit and expiration.
+ * Commit of {@link Table} to provide {@link CommitMessage} committing.
+ *
+ * <ol>
+ *   <li>Before calling {@link TableCommit#commit}, if user cannot determine if this commit is done
+ *       before, user should first call {@link TableCommit#filterCommitted}.
+ *   <li>Before committing, it will first check for conflicts by checking if all files to be removed
+ *       currently exists, and if modified files have overlapping key ranges with existing files.
+ *   <li>After that it use the external {@link CatalogLock} (if provided) or the atomic rename of
+ *       the file system to ensure atomicity.
+ *   <li>If commit fails due to conflicts or exception it tries its best to clean up and aborts.
+ *   <li>If atomic rename fails it tries again after reading the latest snapshot from step 2.
+ * </ol>
```

Comment:
I think we can leave these comments:
```
 * <ol>
 *   <li>Before calling {@link TableCommit#commit}, if user cannot determine if this commit is done
 *       before, user should first call {@link TableCommit#filterCommitted}.
 *   <li>Commit may fail and throw an exception. Before committing, it will first check for conflicts by checking if all files to be removed currently exists.
 * </ol>
```